### DISCUSSION TITLE
Globally set variable type by its name

### DISCUSSION THREAD

**Discussion by macanton** [castwide/solargraph#564] [General]
Is it possible to set globally variable type by name? 
For example, set that the `user` variable is always an instance of the `User` class?

**Comment by castwide** [comment]
If you want it to be truly global, you could define it with a `@!method` directive in the `Kernel` module. Example:

```ruby
module Kernel
  # @!method user
  #   @return [User]
end

user # recognized here

class Example
  user # and here

  def foo
    user # and here
  end

  def self.bar
    user # and here
  end
end
```
